torch_np/_reductions.py

```diff
@@ -198,48 +280,75 @@ def var(tensor, axis=None, dtype=None, ddof=0, *, where=None):
 #   2. axis=None flattens
 
 
-def cumprod(tensor, axis, dtype=None):
+def cumsum(
+    a: ArrayLike,
+    axis: AxisLike = None,
+    dtype: Optional[DTypeLike] = None,
+    out: Optional[OutArray] = None,
+):
     if dtype == torch.bool:
         dtype = _dtypes_impl.default_dtypes.int_dtype
     if dtype is None:
-        dtype = tensor.dtype
+        dtype = a.dtype
 
-    (tensor,), axis = _util.axis_none_flatten(tensor, axis=axis)
-    axis = _util.normalize_axis_index(axis, tensor.ndim)
+    (a,), axis = _util.axis_none_flatten(a, axis=axis)
+    axis = _util.normalize_axis_index(axis, a.ndim)
 
-    return tensor.cumprod(axis=axis, dtype=dtype)
+    return a.cumsum(axis=axis, dtype=dtype)
 
 
-def cumsum(tensor, axis, dtype=None):
+def cumprod(
+    a: ArrayLike,
+    axis: AxisLike = None,
+    dtype: Optional[DTypeLike] = None,
+    out: Optional[OutArray] = None,
+):
     if dtype == torch.bool:
         dtype = _dtypes_impl.default_dtypes.int_dtype
     if dtype is None:
-        dtype = tensor.dtype
+        dtype = a.dtype
 
-    (tensor,), axis = _util.axis_none_flatten(tensor, axis=axis)
-    axis = _util.normalize_axis_index(axis, tensor.ndim)
+    (a,), axis = _util.axis_none_flatten(a, axis=axis)
+    axis = _util.normalize_axis_index(axis, a.ndim)
 
-    return tensor.cumsum(axis=axis, dtype=dtype)
+    return a.cumprod(axis=axis, dtype=dtype)
 
 
-def average(a, axis, weights, returned=False, keepdims=False):
+cumproduct = cumprod
+
+
+@_deco_axis_expand
+def average(
+    a: ArrayLike,
+    axis=None,
+    weights: ArrayLike = None,
+    returned=False,
+    *,
+    keepdims=False,
```

Comment:
This is not type annotated to avoid the generic processing of `keepdims` because of the variadic return, correct? Might want to add a code comment.